---
title: C6255 | Documents Microsoft
ms.custom: ''
ms.date: 11/04/2016
ms.technology:
- vs-ide-code-analysis
ms.topic: conceptual
f1_keywords:
- C6255
helpviewer_keywords:
- C6255
ms.assetid: bb6430b2-782a-4410-a8e1-609df06007de
author: mikeblome
ms.author: mblome
manager: douge
ms.workload:
- multiple
ms.openlocfilehash: e9374b6c0a8d1ede3c0b7fe6d3ef01e71c2715fb
ms.sourcegitcommit: 6a9d5bd75e50947659fd6c837111a6a547884e2a
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 04/16/2018
---
# <a name="c6255"></a>C6255
Avertissement C6255 : _alloca signale un échec en levant une exception de dépassement de capacité de pile. Utilisez _malloca à la place  
  
 Cet avertissement indique qu’un appel à `_alloca` a été détecté en dehors de la gestion des exceptions locales. `_alloca` doit toujours être appelée à partir de la plage protégée d’un gestionnaire d’exceptions car il peut lever une exception de dépassement de capacité de pile en cas d’échec. Si possible, au lieu d’utiliser `_alloca`, envisagez d’utiliser `_malloca` qui est une version plus sécurisée de `_alloca`.  
  
## <a name="example"></a>Exemple  
 Le code suivant génère cet avertissement car `_alloca` peut générer l’exception :  
  
```  
#include <windows.h>  
  
void f( )  
{  
  void *p = _alloca(10);  
  // code ...  
}  
  
```  
  
 Pour corriger cet avertissement, utilisez `_malloca` et ajoutez le Gestionnaire d’exceptions comme indiqué dans le code suivant :  
  
```  
#include <windows.h>  
#include <malloc.h>  
  
void f( )  
{  
  void *p;  
  int errcode;  
  __try   
  {  
    p = _malloca(10);  
    // code...  
    _freea(p);  
  }  
  __except( (GetExceptionCode() == STATUS_STACK_OVERFLOW ) ?  
              EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )  
  {  
    errcode = _resetstkoflw();  
    // code ...  
  }  
}  
```  
  
## <a name="see-also"></a>Voir aussi  
 [_malloca](/cpp/c-runtime-library/reference/malloca)